### PR TITLE
Add linting workflow

### PR DESCRIPTION
**Pull request by nathanchance**
I am not sure how this will feel but we do it on all of our other repos, I think that it is nice for consistency.

### CODE REVIEW

File: generate_workflow.py [hunk 1]
```diff
@@ -43,23 +46,33 @@ def get_steps(build):
...
         sanitize_job_name(name): {
-            "runs-on": "ubuntu-20.04",
-            "needs": "kick_tuxsuite",
-            "name": name,
+            "runs-on":
+            "ubuntu-20.04",
+            "needs":
+            "kick_tuxsuite",
+            "name":
+            name,
```
Comment (nickdesaulniers) [inline generate_workflow.py:54]: this change "hurts" but formatter knows best, I suppose...

**Review by nickdesaulniers — APPROVED** [review]
thanks! 🧼 🐌

File: generate_workflow.py (hunk 1 above)
Comment (nathanchance) [reply generate_workflow.py:54]: I can add `# yapf: disable` and `# yapf: enable` if we want to keep that the way it is for readability?

Comment (nickdesaulniers) [reply generate_workflow.py:54]: sure, at the least we won't have to go look up the syntax for that every time we want to overrule the linter.

**Comment by nickdesaulniers** [comment]
lgtm

**Pull request merged**